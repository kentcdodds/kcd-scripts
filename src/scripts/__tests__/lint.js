import cases from 'jest-in-case'
import {unquoteSerializer, winPathSerializer} from './helpers/serializers'

jest.mock('jest', () => ({run: jest.fn()}))
jest.mock('../../config/jest.config', () => ({builtInConfig: true}))

expect.addSnapshotSerializer(unquoteSerializer)
expect.addSnapshotSerializer(winPathSerializer)

cases(
  'format',
  ({
    args = [],
    utils = require('../../utils'),
    hasPkgProp = () => false,
    hasFile = () => false,
    setup = () => () => {},
  }) => {
    // beforeEach
    const {sync: crossSpawnSyncMock} = require('cross-spawn')
    const originalArgv = process.argv
    const originalExit = process.exit
    Object.assign(utils, {
      hasPkgProp,
      hasFile,
      resolveBin: (modName, {executable = modName} = {}) => executable,
    })
    process.exit = jest.fn()
    const teardown = setup()

    process.argv = ['node', '../lint', ...args]
    crossSpawnSyncMock.mockClear()

    try {
      // tests
      require('../lint')
      expect(crossSpawnSyncMock).toHaveBeenCalledTimes(1)
      const [firstCall] = crossSpawnSyncMock.mock.calls
      const [script, calledArgs] = firstCall
      expect([script, ...calledArgs].join(' ')).toMatchSnapshot()
    } catch (error) {
      throw error
    } finally {
      teardown()
      // afterEach
      process.exit = originalExit
      process.argv = originalArgv
      jest.resetModules()
    }
  },
  {
    'calls eslint CLI with default args': {},
    'does not use built-in config with --config': {
      args: ['--config', './custom-config.js'],
    },
    'does not use built-in config with .eslintrc file': {
      hasFile: filename => filename === '.eslintrc',
    },
    'does not use built-in config with .eslintrc.js file': {
      hasFile: filename => filename === '.eslintrc.js',
    },
    'does not use built-in config with eslintConfig pkg prop': {
      hasPkgProp: prop => prop === 'eslintConfig',
    },
    'does not use built-in ignore with --ignore-path': {
      args: ['--ignore-path', './my-ignore'],
    },
    'does not use built-in ignore with .eslintignore file': {
      hasFile: filename => filename === '.eslintignore',
    },
    'does not use built-in ignore with eslintIgnore pkg prop': {
      hasPkgProp: prop => prop === 'eslintIgnore',
    },
    '--no-cache will disable caching': {
      args: ['--no-cache'],
    },
    'runs on given files, but only js files': {
      args: [
        './src/index.js',
        './package.json',
        './src/index.css',
        './src/component.js',
      ],
    },
  },
)
