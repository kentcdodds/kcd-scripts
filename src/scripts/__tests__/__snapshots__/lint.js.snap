// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint --no-cache will disable caching 1`] = `eslint --config ./src/config/eslintrc.js --ignore-path ./src/config/eslintignore --no-cache .`;

exports[`lint calls eslint CLI with default args 1`] = `eslint --config ./src/config/eslintrc.js --ignore-path ./src/config/eslintignore --cache .`;

exports[`lint does not use built-in config with .eslintrc file 1`] = `eslint --ignore-path ./src/config/eslintignore --cache .`;

exports[`lint does not use built-in config with .eslintrc.js file 1`] = `eslint --ignore-path ./src/config/eslintignore --cache .`;

exports[`lint does not use built-in config with --config 1`] = `eslint --ignore-path ./src/config/eslintignore --cache --config ./custom-config.js .`;

exports[`lint does not use built-in config with eslintConfig pkg prop 1`] = `eslint --ignore-path ./src/config/eslintignore --cache .`;

exports[`lint does not use built-in ignore with .eslintignore file 1`] = `eslint --config ./src/config/eslintrc.js --cache .`;

exports[`lint does not use built-in ignore with --ignore-path 1`] = `eslint --config ./src/config/eslintrc.js --cache --ignore-path ./my-ignore .`;

exports[`lint does not use built-in ignore with eslintIgnore pkg prop 1`] = `eslint --config ./src/config/eslintrc.js --cache .`;

exports[`lint runs on given files, but only js files 1`] = `eslint --config ./src/config/eslintrc.js --ignore-path ./src/config/eslintignore --cache ./src/index.js ./src/component.js`;
